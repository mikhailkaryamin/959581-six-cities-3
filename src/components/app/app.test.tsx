import * as React from 'react';
import * as renderer from 'react-test-renderer';
import App from './app';
import configureStore from 'redux-mock-store';
import {
  Provider
} from 'react-redux';
import {
  Store,
} from '../../mocks/initialStateTest';

const mockStore = configureStore([]);

describe(`App render correctly`, () => {
  test(`App render correctly AUTH`, () => {
    const store = mockStore(Store.WITH_AUTH);
    const tree = renderer
      .create(
          <Provider
            store={store}
          >
            <App />
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test(`App render correctly NO AUTH`, () => {
    const store = mockStore(Store.NO_AUTH);
    const tree = renderer
      .create(
          <Provider
            store={store}
          >
            <App />
          </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
