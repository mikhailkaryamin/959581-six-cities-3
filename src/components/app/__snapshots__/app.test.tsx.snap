// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render correctly App render correctly AUTH 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/959581-six-cities-3/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="../img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/959581-six-cities-3/favorite/"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                  style={
                    Object {
                      "backgroundImage": "url(https://htmlacademy-react-3.appspot.com/six-cities)",
                    }
                  }
                />
                <span
                  className="header__user-name user__name"
                >
                  mail@mail.ru
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  
  
</div>
`;

exports[`App render correctly App render correctly NO AUTH 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/959581-six-cities-3/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="../img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/959581-six-cities-3/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  
  
</div>
`;
